code/Controllers/CMSMain.php

```diff
@@ -2265,7 +2233,7 @@ public function restore($data, $form)
         return $this->getResponseNegotiator()->respond($this->getRequest());
     }
 
-    public function duplicate($request)
+    public function duplicate(HTTPRequest $request): HTTPResponse
```

Comment:
This method sometimes returns `Security::permissionFailure()` which sometimes returns an array

Reply:
As noted above, it does not return an array